src/core/ext/filters/client_channel/client_channel.cc

```diff
@@ -1435,18 +1435,15 @@ ChannelData::ChannelData(grpc_channel_element_args* args, grpc_error** error)
       std::move(target_uri), ProcessResolverResultLocked, this, error));
   grpc_channel_args_destroy(new_args);
   if (*error != GRPC_ERROR_NONE) {
-    // Orphan the resolving LB policy and flush the exec_ctx to ensure
-    // that it finishes shutting down.  This ensures that if we are
-    // failing, we destroy the ClientChannelControlHelper (and thus
-    // unref the channel stack) before we return.
+    // Before we return, shut down the resolving LB policy, which destroys
+    // the ClientChannelControlHelper and therefore unrefs the channel stack.
     // TODO(roth): This is not a complete solution, because it only
     // catches the case where channel stack initialization fails in this
     // particular filter.  If there is a failure in a different filter, we
     // will leave a dangling ref here, which can cause a crash.  Fortunately,
     // in practice, there are no other filters that can cause failures in
     // channel stack initialization, so this works for now.
     resolving_lb_policy_.reset();
```

Comment:
Why are we already in the combiner here? I forgot it..

Reply:
We're not technically in the combiner here, but it doesn't matter, because if we fail this early (we're still in the ChannelData ctor), there can't possibly be anything happening in any other thread yet.

Reply:
I see..